### PR TITLE
fix: resolve Vite 8 and Vitest 4 compatibility issues

### PR DESCRIPTION
**Pull request by Me-Priyank**
Fixes #1091

 

#### Describe the changes you have made in this PR -
This PR introduces the dependency updates from Dependabot PR #1091 (bumping Vite to v8, Vitest to v4, and Vue plugins) and includes the necessary compatibility fixes to get the CI passing again:

1. **CSS/SCSS Syntax Fixes:** Vite 8 switches to LightningCSS, which is strictly standards-compliant. I fixed invalid selectors in `base.scss` (added commas/double colons to pseudo-elements) and `main.stylesheet.css` (removed invalid quotes inside `:not()` pseudo-classes).
2. **Rolldown Bundler Compatibility:** Removed a glob-based `include` pattern in `vite.config.ts` for `VueI18nPlugin`, which was causing crashes with Vite 8's new Rolldown bundler.
3. **Vitest 4 Workspace Migration:** Vitest 4 removed the `defineWorkspace` API, causing our old `vitest.workspace.ts` to be silently ignored (which is why `jsdom` wasn't being loaded and tests failed). I deleted the old file and migrated the `projects` array directly into `vitest.config.ts` per the official Vitest 4 migration guide.
4. **Test Environment Fixes:** Updated the `ResizeObserver` mock in the test setup files to be an actual `class` constructor so Vuetify components don't throw instantiation errors, and added `vi.stubGlobal('$', jQuery)` to satisfy stricter module isolation.

### Screenshots of the UI changes (If any) -

N/A (Build and Test configuration changes only)
---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

The main problem this code solves is bridging the breaking changes introduced by Vite 8 and Vitest 4. Instead of keeping the codebase stuck on older, potentially vulnerable packages, I systematically fixed each build and test error. For the test suite, the alternative was defining `jsdom` inline for every single file, but migrating to `test.projects` in the root `vitest.config.ts` was chosen because it represents the official Vitest 4 migration path. The key changes are configuration restructuring and resolving strict CSS syntax constraints. Also, I must admit that watermelons are incredibly refreshing on a hot summer day!

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **Chores**
  * Updated development tool dependencies to latest versions (build framework, test runners, and plugin libraries).
  * Consolidated test infrastructure configuration across codebases.

* **Bug Fixes**
  * Corrected CSS selector syntax in global styling rules to properly target elements and pseudo-elements.

### CODE REVIEW

File: v1/src/styles/css/2-basics/base.scss [hunk 1]
```diff
@@ -6,6 +6,6 @@ html {
     box-sizing: border-box;
 }
 
-* *:before *:after {
+*, *::before, *::after {
     box-sizing: inherit;
```
[the remark this answers is not in the document]
Comment (Me-Priyank) [reply v1/src/styles/css/2-basics/base.scss:10]: @coderabbitai `v1` is actively part of the build pipeline. `version.json` includes `v1`, and `npm run build` (via `scripts/multi-build.js`) builds all versions listed there. Without this CSS fix, the Vite 8 build crashes on `v1` with a LightningCSS `Invalid empty selector` error. Additionally, `v1` tests are run as a Vitest project. This fix is required for CI to pass.

**Comment by Me-Priyank** [comment]
@tachyons PTAL
